frontend/src/app/(protected)/experiments/api.ts

```diff
@@ -1,15 +1,29 @@
 import api from "@/utils/api";
-import { NewMAB } from "./types";
+import { MABExperimentState, ABExperimentState } from "./types";
+import { ExperimentState } from "./types";
 
-const createMABExperiment = async ({
-  mab,
+const createNewExperiment = async ({
+  experimentData,
   token,
 }: {
-  mab: NewMAB;
+  experimentData: ExperimentState;
   token: string | null;
 }) => {
+  let endpoint: string;
+  let newExperimentData: MABExperimentState | ABExperimentState;
+
+  if (experimentData.methodType == "mab") {
```

Comment:
Another place where you need to update when we have a new method.
I don't like it. Need to ideally abstract methods away from the rest of the code. 

Will fix in a future PR.